Kudu.Core/Extensions/JobManagerBaseIReadOnlyCollectionExtensions.cs

```diff
@@ -0,0 +1,54 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using Kudu.Contracts.Jobs;
+using Kudu.Contracts.Settings;
+
+namespace Kudu.Core.Extensions
+{
+    public static class JobManagerBaseIReadOnlyCollectionExtensions
+    {
+        public static IJobsManager<T> GetPrimaryManager<T>(this IReadOnlyCollection<IJobsManager<T>> collection) where T : JobBase, new()
+        {
+            return collection.FirstOrDefault(m => m.BasePathType == WebJobsBasePathType.Primary);
+        }
+
+        public static IJobsManager<T> GetSecondaryManager<T>(this IReadOnlyCollection<IJobsManager<T>> collection) where T : JobBase, new()
+        {
+            return collection.FirstOrDefault(m => m.BasePathType == WebJobsBasePathType.Secondary);
+        }
+
+        public static IContinuousJobsManager GetWriteManager(this IReadOnlyCollection<IContinuousJobsManager> collection, IDeploymentSettingsManager settings)
+        {
+            return settings.RunFromLocalZip() || settings.RunFromRemoteZip()
+                ? collection.GetSecondaryManager() as IContinuousJobsManager
+                : collection.GetPrimaryManager() as IContinuousJobsManager;
+        }
+
+        public static ITriggeredJobsManager GetWriteManager(this IReadOnlyCollection<ITriggeredJobsManager> collection, IDeploymentSettingsManager settings)
+        {
+            return settings.RunFromLocalZip() || settings.RunFromRemoteZip()
+                ? collection.GetSecondaryManager() as ITriggeredJobsManager
+                : collection.GetPrimaryManager() as ITriggeredJobsManager;
+        }
+
+        /// <summary>
+        /// Jobs in primary manager take precedence over other manager.
+        /// </summary>
+        /// <param name="collection"></param>
+        /// <param name="jobName"></param>
+        /// <returns></returns>
+        public static ITriggeredJobsManager GetManagerForJob(this IReadOnlyCollection<ITriggeredJobsManager> collection, string jobName)
+        {
+            return collection.GetPrimaryManager().HasJob(jobName)
+                ? collection.GetPrimaryManager() as ITriggeredJobsManager
+                : collection.GetSecondaryManager() as ITriggeredJobsManager;
+        }
+
+        public static IContinuousJobsManager GetManagerForJob(this IReadOnlyCollection<IContinuousJobsManager> collection, string jobName)
+        {
+            return collection.GetPrimaryManager().HasJob(jobName)
+                ? collection.GetPrimaryManager() as IContinuousJobsManager
+                : collection.GetSecondaryManager() as IContinuousJobsManager;
+        }
```

Comment:
Have you considered an AggregrateContinuousJobsManager : IContinuousJobsManager?   This job managers will wrap the collection of JobManagers and delegate call to them as neccessary.

Reply:
Thank you, that's a much better idea. I updated the PR.